common-lib/src/cmr/common/validations/json_schema.clj

```diff
@@ -7,102 +7,64 @@
-  the report. Returns nil if there are no errors. Takes a
-  com.github.fge.jsonschema.core.report.ListProcessingReport.
-
-  For details, see:
-  * [ProcessingMessage](http://java-json-tools.github.io/json-schema-core/1.2.x/com/github/fge/jsonschema/core/report/ProcessingMessage.html)
-  * [ListProcessingReport](http://java-json-tools.github.io/json-schema-core/1.2.x/com/github/fge/jsonschema/core/report/ListProcessingReport.html)"
-  [^ListProcessingReport report]
-  (flatten
-    (for [error-report (seq (.asJson report))
-          :let [json-error-report (json/decode (str ^Object error-report) true)]]
-      (conj (parse-nested-error-report (:reports json-error-report))
-            (parse-error-report json-error-report)))))
-
-(defn- json-string->JsonNode
-  "Takes JSON as a string or as EDN and returns a
-  com.fasterxml.jackson.databind.JsonNode. Throws an exception if the provided
-  JSON is not valid JSON."
+(defn- json-string->JSONObject
   [json-string]
+  "Takes JSON as a string and returns a org.json.JSONObject.
+   Throws an exception if the provided JSON is not valid JSON."
   (try
-    (JsonLoader/fromString json-string)
-    (catch JsonParseException e
-      ;; Removes the source from the message which is just going to say
-      ;; StringReader@XXXXX
-      (let [message (str/replace (.getMessage e) #"\[Source[^;]+;" "")]
-        (errors/throw-service-error :bad-request (str "Invalid JSON: " message))))))
+    (JSONObject. (JSONTokener. json-string))
+    (catch JSONException e
+      (errors/throw-service-error :bad-request (str "Invalid JSON: " (.getMessage e))))))
 
 (defn json-string->json-schema
-  "Convert a string to com.github.fge.jsonschema.main.JsonSchema object."
+  "Convert a string to org.everit.json.schema.Schema object."
   [schema-string]
-  (->> schema-string
-       JsonLoader/fromString
-       (.getJsonSchema (JsonSchemaFactory/byDefault))))
+  (SchemaLoader/load (json-string->JSONObject schema-string)))
 
 (defn parse-json-schema
   "Convert a Clojure object to a JSON string then parses into a
-  com.github.fge.jsonschema.main.JsonSchema object."
+  org.everit.json.schema.Schema object."
   [schema-def]
   (->> (assoc schema-def :$schema "http://json-schema.org/draft-04/schema#")
        json/generate-string
        json-string->json-schema))
 
 (defn parse-json-schema-from-uri
-  "Loads a JSON schema from a URI into a com.github.fge.jsonschema.main.JsonSchema
+  "Loads a JSON schema from a URI into a org.everit.json.schema.Schema
   object. It's necessary to use this one if loading a JSON schema from the
   classpath that references other JSON schemas on the classpath."
   [uri]
-  (let [factory (JsonSchemaFactory/byDefault)]
-    (.getJsonSchema factory (str uri))))
+  (let [raw-schema (json-string->JSONObject (slurp (str uri)))]
+    (-> (SchemaLoader/builder)
+        (.schemaJson raw-schema)
+        (.resolutionScope (str uri))
+        .build
```

Comment:
`.build` twice?

Reply:
Yeah so the first called build() returns a SchemaLoader object then the call to load() returns a Schema.Builder() object and finally the last call to build() returns a Schema Object.

So it goes:
SchemaLoader.SchemaLoaderBuilder -> SchemaLoader -> Schema.Builder -> Schema.

Reply:
Might be worth leaving a comment to explain why it was done this way to prevent confusion in the future